### PR TITLE
iOS fix

### PR DESCRIPTION
**Pull request by Mehrdadmaskull**
Updates to the Google Maps iOS SDK and fixing the current location issue

### CODE REVIEW

File: ios/Runner.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -17,6 +17,7 @@
 		97C146FC1CF9000F007C117D /* Main.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 97C146FA1CF9000F007C117D /* Main.storyboard */; };
 		97C146FE1CF9000F007C117D /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 97C146FD1CF9000F007C117D /* Assets.xcassets */; };
 		97C147011CF9000F007C117D /* LaunchScreen.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 97C146FF1CF9000F007C117D /* LaunchScreen.storyboard */; };
+		EE91B99237E91BD5F7ED765C /* Pods_Runner.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = 36786F88E51B42C1CE027EAC /* Pods_Runner.framework */; };
```
Comment (RumianR) [inline ios/Runner.xcodeproj/project.pbxproj]: Is this file necessary? It seems to be something that gets built from the user side.

**Comment by MrJCipherButtles** [comment]
@Mehrdadmaskull I pulled your branch and tried running the app on my iphone and the build failed and the app never launches? Last I have heard from Naimur, is that he too still cannot build the app on his iphone. Your pull request seems to fail at actually resolving the ios build issue. Are there steps we are missing that need to be performed manually? You should list how exactly you got the app to run on your iphone. Thanks

File: ios/Runner.xcodeproj/project.pbxproj (hunk 1 above)
Comment (Mehrdadmaskull) [reply ios/Runner.xcodeproj/project.pbxproj]: Even though it gets generated each time, it's better to keep it there as a base, so if someone doesn't have the minimum version, it would not run unless they have the latest changes on the repo.

**Comment by Mehrdadmaskull** [comment]
> @Mehrdadmaskull I pulled your branch and tried running the app on my iphone and the build failed and the app never launches? Last I have heard from Naimur, is that he too still cannot build the app on his iphone. Your pull request seems to fail at actually resolving the ios build issue. Are there steps we are missing that need to be performed manually? You should list how exactly you got the app to run on your iphone. Thanks

Only those who have the iOS dev account and have set their account on Xcode can run the app on their phone, unless your iOS is less than 13.3.

**Comment by MrJCipherButtles** [comment]
I still can't get it to run, I can run the basic flutter counter app on my iphone. First I tried running directly from your branch, it did not run, and saw that it was hard coded to your development team account, so, I changed that, but still I get this error:
`An error was encountered while attempting to communicate with this device. (The service is invalid.)
Domain: com.apple.dtdevicekit
Code: -402653150
Failure Reason: Please try rebooting and reconnecting the device. (0xE8000022).
User Info: {
    DVTRadarComponentKey = 487927;
    "com.apple.dtdevicekit.stacktrace" = (
	0   DTDeviceKitBase                     0x0000000127ce16e7 DTDKCreateNSError + 109
	1   DTDeviceKitBase                     0x0000000127ce1de9 DTDK_AMDErrorToNSError + 792
	2   DTDeviceKitBase                     0x0000000127cf2d0c __63-[DTDKRemoteDeviceConnection startFirstServiceOf:unlockKeybag:]_block_invoke + 301
	3   DTDeviceKitBase                     0x0000000127cf24db __48-[DTDKRemoteDeviceConnection futureWithSession:]_block_invoke_3 + 22
	4   DTDeviceKitBase                     0x0000000127ce415f __DTDKExecuteInSession_block_invoke_2 + 35
	5   DTDeviceKitBase                     0x0000000127ce3593 __DTDKExecuteWithConnection_block_invoke_2 + 473
	6   DTDeviceKitBase                     0x0000000127ce3391 __DTDKExecuteWithConnection_block_invoke + 106
	7   libdispatch.dylib                   0x00007fff7311950e _dispatch_client_callout + 8
	8   libdispatch.dylib                   0x00007fff73125567 _dispatch_lane_barrier_sync_invoke_and_complete + 60
	9   DVTFoundation                       0x000000010f505110 DVTDispatchBarrierSync + 208
	10  DVTFoundation                       0x000000010f4dc538 -[DVTDispatchLock performLockedBlock:] + 64
	11  DTDeviceKitBase                     0x0000000127ce3292 DTDKExecuteWithConnection + 226
	12  DTDeviceKitBase                     0x0000000127ce3fc9 DTDKExecuteInSession + 376
	13  DTDeviceKitBase                     0x0000000127cf2410 __48-[DTDKRemoteDeviceConnection futureWithSession:]_block_invoke_2 + 112
	14  DVTFoundation                       0x000000010f502e7a __DVT_CALLING_CLIENT_BLOCK__ + 7
	15  DVTFoundation                       0x000000010f504552 __DVTDispatchAsync_block_invoke + 809
	16  libdispatch.dylib                   0x00007fff73118583 _dispatch_call_block_and_release + 12
	17  libdispatch.dylib                   0x00007fff7311950e _dispatch_client_callout + 8
	18  libdispatch.dylib                   0x00007fff7311eace _dispatch_lane_serial_drain + 597
	19  libdispatch.dylib                   0x00007fff7311f452 _dispatch_lane_invoke + 363
	20  libdispatch.dylib                   0x00007fff73128a9e _dispatch_workloop_worker_thread + 598
	21  libsystem_pthread.dylib             0x00007fff733736fc _pthread_wqthread + 290
	22  libsystem_pthread.dylib             0x00007fff73372827 start_wqthread + 15
);
}
--
`

**Comment by Mehrdadmaskull** [comment]
@MrJCipherButtles like I said you need a developer account to run it. You can't use your Apple ID.

**Comment by MrJCipherButtles** [comment]
568019E8-96F0-44D6-8E4D-76D8CC72CE58_4_5005_c

I have an apple development account...

**Comment by MrJCipherButtles** [comment]
@Mehrdadmaskull 

I took it upon myself to submit an appropriate PR #99 dealing with the "iOS fix". Please take note of proper practices by reviewing that PR. The iOS fix was about getting people to build the project on their iPhone's, there were many requirements that you should have made known with your PR. Also, the actual "code fix" was literally 2 lines in Info.plist. This should have been made transparent and your focus should have been on explaining how to setup.

This PR has been a mess of confusion from the beginning, you have 25 commits for what is suppose to be a simple iOS fix, and on top of that you started adding development commits that are unrelated to the PR. I'm sorry that this may not come as a welcoming comment to you, but it really is intended for the best. When you submit a PR, people actually need to review the changes your work is bringing into master, and you need to manage your PRs better.

Even as you started adding commits for the direction's feature, your changes show additions (green lines of code) for code that is already in master? For example, the equals method in Coordinate.dart. And, there are many more green chunks of code that falsely show up like additions to master. I don't even understand how this is possible. You are severely misusing revert commits.

Also, PRs #91 , #92 , and this PR were all created by you for the same pull request. You make one PR and if someone request's changes you do not delete your PR and submit a new one. You are suppose to simply update the same PR with your changes and re-request reviews once you are done making changes.

I'm sorry, but this simply cannot keep going on. Please talk to someone in the team if you need more guidance with GitHub.

I'm sorry if you feel like I've put you on the spot. But this can also serve others as an example of what not to do when submitting PRs and things that they should watch out for to assist with reviews so that the PR process can be a streamlined process. I'm sure you are not the only one with the above mentioned difficulties, so, don't feel singled out by this.

Respectfully,

Josh

**Comment by MrJCipherButtles** [comment]
This PR has been resolved with #99 ; therefore, I am closing this
